core/src/main/scala/org/apache/spark/Partitioner.scala

```diff
@@ -56,7 +56,7 @@ object Partitioner {
    */
   def defaultPartitioner(rdd: RDD[_], others: RDD[_]*): Partitioner = {
     val bySize = (Seq(rdd) ++ others).sortBy(_.partitions.size).reverse
-    for (r <- bySize if r.partitioner.isDefined) {
+    for (r <- bySize if r.partitioner.isDefined && r.partitions.nonEmpty) {
       return r.partitioner.get
     }
```

Comment:
The for comprehension over the Option effectively does the same thing as an explicit `r.partitioner.isDefined` -- nothing will be returned if `r.partitioner` is `None`.  This is just idiomatic Scala.